pkg/storages/s3/folder.go

```diff
@@ -278,10 +283,20 @@ func (folder *Folder) listObjectsPagesV1(prefix *string, delimiter *string,
 		Prefix:    prefix,
 		Delimiter: delimiter,
 	}
-	return folder.S3API.ListObjectsPages(s3Objects, func(files *s3.ListObjectsOutput, lastPage bool) bool {
-		listFunc(files.CommonPrefixes, files.Contents)
-		return true
-	})
+	uploadTimeout := folder.getUploadTimeout()
+	if uploadTimeout != 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Duration(uploadTimeout * int(time.Second)))
+		defer cancel()
+		return folder.S3API.ListObjectsPagesWithContext(ctx, s3Objects, func(files *s3.ListObjectsOutput, lastPage bool) bool {
```

Comment:
The only difference is the context, so we can just alter this part instead of duplicating the entire call to `folder.S3API.ListObjectsPages`. I mean, for no timeout we can just pass `context.Background()` and `context.WithTimeout()` in other case.

Reply:
Also, I see that upload timeout affects not only the upload part but also the listings. Are you sure about the naming?

Reply:
there is also difference in called functions: ListObjectsPagesWithContext and ListObjectsV2PagesWithContext

about namings: yes, I'll change it